app/src/main/java/org/fitchfamily/android/dejavu/BackendService.kt

```diff
@@ -513,7 +518,14 @@ class BackendService : LocationBackendService() {
         }
         val mcc = mncString.substring(0, 3).toIntOrNull() ?: return observations
         val mnc = mncString.substring(3).toIntOrNull() ?: return observations
-        val info = telephonyManager!!.cellLocation
+        var info: CellLocation? = null
+
+        try {
+            val info = telephonyManager!!.cellLocation
+        } catch (e: Throwable) {
+            if (DEBUG) Log.d(TAG, "getCellLocation(): failed")
```

Comment:
What was the actual crash you saw here?

Reply:
```
FATAL EXCEPTION: DefaultDispatcher-worker-6
Process: helium314.localbackend, PID: 8445
java.lang.NullPointerException: Attempt to invoke virtual method 'android.telephony.CellLocation android.telephony.CellIdentity.asCellLocation()' on a null object reference
	at android.telephony.TelephonyManager.getCellLocation(TelephonyManager.java:2339)
	at org.fitchfamily.android.dejavu.BackendService.deprecatedGetMobileTowers(BackendService.kt:519)
	at org.fitchfamily.android.dejavu.BackendService.getMobileTowers(BackendService.kt:385)
	at org.fitchfamily.android.dejavu.BackendService.scanMobile(BackendService.kt:356)
	at org.fitchfamily.android.dejavu.BackendService.access$scanMobile(BackendService.kt:46)
	at org.fitchfamily.android.dejavu.BackendService$startMobileScan$1.invokeSuspend(BackendService.kt:346)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
```

Reply:
In my radiotracker I've used `telephonyManager.getAllCellInfo` as getCellLocation is deprecated and crashing

Reply:
`getAllCellInfo` is used by default, `getCellLocation` is just a fallback that is necessary on some older phones.

So basically `TelephonyManager.getCellLocation` always crashes on your device?
It's rather bad that we need to catch this here, but it definitely needs to be done.